src/tests/integration/test_foundry_prove.py

```diff
@@ -334,6 +334,65 @@ def test_foundry_merge_nodes(
     assert_pass(test, single(prove_res))
 
 
+def test_foundry_merge_loop_heads(
+    foundry: Foundry,
+    update_expected_output: bool,
+    bug_report: BugReport | None,
+    server: KoreServer,
+    no_use_booster: bool,
+) -> None:
+    test = 'BMCLoopsTest.test_bmc(uint256)'
+
+    if no_use_booster and test in SKIPPED_LEGACY_TESTS:
+        pytest.skip()
+
+    if bug_report is not None:
+        server._populate_bug_report(bug_report)
+
+    foundry_prove(
+        foundry,
+        tests=[(test, None)],
+        prove_options=ProveOptions(
+            max_iterations=20,
+            bug_report=bug_report,
+        ),
+        rpc_options=RPCOptions(
+            port=server.port,
+        ),
+    )
+
+    foundry_merge_nodes(foundry, test=test, node_ids=[15, 16], include_disjunct=True)
+
+    foundry_prove(
+        foundry,
+        tests=[(test, None)],
+        prove_options=ProveOptions(
+            max_iterations=10,
+            bug_report=bug_report,
+        ),
```

Comment:
```suggestion
        prove_options=ProveOptions(
            max_iterations=2,
            bug_report=bug_report,
        ),
```

Just to reduce how long this test runs for, and how much output is generated, can you restrict this to just 2 iterations after the node merge, and then update the expected output?